File: 485. Max Consecutive Ones.go
func findMaxConsecutiveOnes(nums []int) int {
    var cnt = 0
    var maxcnt = 0
    var l = len(nums)
    for i := 0; i < l; i++ {
        if nums[i] == 1 {
            cnt++
        } else {
            if cnt > maxcnt {
                maxcnt = cnt
            }
            cnt = 0
        }
    }
    if cnt > maxcnt {
        maxcnt = cnt
    }
    return maxcnt
}

File: 485. Max Consecutive Ones.rb
# @param {Integer[]} nums
# @return {Integer}
def find_max_consecutive_ones(nums)
    cnt = 0
    maxcnt = 0
    for i in 0..nums.size
        if nums[i] == 1
            cnt = cnt + 1
        else
            maxcnt = (maxcnt > cnt ? maxcnt : cnt)
            cnt = 0
        end
    end
    return maxcnt > cnt ? maxcnt : cnt
end
